### PR TITLE
Downgrade K8s dependency version to v0.24

### PR DESCRIPTION
**Pull request by sedefsavas**
/kind bug

**What this PR does / why we need it**:
Using k8s v0.25.x dependencies requires go version 1.19, this pr downgrades the k8s dependency versions to v0.24.2 which is same with CAPI v1.2.2.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Checklist**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests
- [ ] adds or updates e2e tests

### CODE REVIEW

**Comment by sedefsavas** [comment]
cc @Skarlso

File: main.go [hunk 1]
```diff
@@ -115,7 +114,7 @@ func main() {
 	pflag.CommandLine.AddGoFlagSet(flag.CommandLine)
 	pflag.Parse()
 
-	if err := logsv1.ValidateAndApply(logOptions, nil); err != nil {
+	if err := logOptions.ValidateAndApply(nil); err != nil {
```
Comment (Skarlso) [inline main.go:117]: Is this right? I thought I made a mistake here and the fix was to use that logv1 package instead.

Comment (sedefsavas) [reply main.go:117]: Yes, logv1 package is in v0.25 and it was causing the bump.

Comment (sedefsavas) [reply main.go:117]: When that line is not removed, go modules pull k8s.io v0.25, you can see this locally.

Comment (Skarlso) [reply main.go:117]: Ah, okay. :)

**Pull request merged**